README.md
# storks

Go to folder requests and follow the readme associated.


requests/HISTORY.md
Release History
===============

dev
---

-   \[Short description of non-trivial change.\]

2.25.1 (2020-12-16)
-------------------

**Bugfixes**

- Requests now treats `application/json` as `utf8` by default. Resolving
  inconsistencies between `r.text` and `r.json` output. (#5673)

**Dependencies**

- Requests now supports chardet v4.x.

2.25.0 (2020-11-11)
------------------

**Improvements**

- Added support for NETRC environment variable. (#5643)

**Dependencies**

- Requests now supports urllib3 v1.26.

**Deprecations**

- Requests v2.25.x will be the last release series with support for Python 3.5.
- The `requests[security]` extra is officially deprecated and will be removed
  in Requests v2.26.0.

2.24.0 (2020-06-17)
-------------------

**Improvements**

- pyOpenSSL TLS implementation is now only used if Python
  either doesn't have an `ssl` module or doesn't support
  SNI. Previously pyOpenSSL was unconditionally used if available.
  This applies even if pyOpenSSL is installed via the
  `requests[security]` extra (#5443)

- Redirect resolution should now only occur when
  `allow_redirects` is True. (#5492)

- No longer perform unnecessary Content-Length calculation for
  requests that won't use it. (#5496)

2.23.0 (2020-02-19)
-------------------

**Improvements**

- Remove defunct reference to `prefetch` in Session `__attrs__` (#5110)

**Bugfixes**

- Requests no longer outputs password in basic auth usage warning. (#5099)

**Dependencies**

- Pinning for `chardet` and `idna` now uses major version instead of minor.
  This hopefully reduces the need for releases everytime a dependency is updated.

2.22.0 (2019-05-15)
-------------------

**Dependencies**

- Requests now supports urllib3 v1.25.2.
  (note: 1.25.0 and 1.25.1 are incompatible)

**Deprecations**

- Requests has officially stopped support for Python 3.4.

2.21.0 (2018-12-10)
-------------------

**Dependencies**

- Requests now supports idna v2.8.

2.20.1 (2018-11-08)
-------------------

**Bugfixes**

- Fixed bug with unintended Authorization header stripping for
  redirects using default ports (http/80, https/443).

2.20.0 (2018-10-18)
-------------------

**Bugfixes**

-   Content-Type header parsing is now case-insensitive (e.g.
    charset=utf8 v Charset=utf8).
-   Fixed exception leak where certain redirect urls would raise
    uncaught urllib3 exceptions.
-   Requests removes Authorization header from requests redirected
    from https to http on the same hostname. (CVE-2018-18074)
-   `should_bypass_proxies` now handles URIs without hostnames (e.g.
    files).

**Dependencies**

- Requests now supports urllib3 v1.24.

**Deprecations**

- Requests has officially stopped support for Python 2.6.

2.19.1 (2018-06-14)
-------------------

**Bugfixes**

-   Fixed issue where status\_codes.py's `init` function failed trying
    to append to a `__doc__` value of `None`.

2.19.0 (2018-06-12)
-------------------

**Improvements**

-   Warn user about possible slowdown when using cryptography version
    &lt; 1.3.4
-   Check for invalid host in proxy URL, before forwarding request to
    adapter.
-   Fragments are now properly maintained across redirects. (RFC7231
    7.1.2)
-   Removed use of cgi module to expedite library load time.
-   Added support for SHA-256 and SHA-512 digest auth algorithms.
-   Minor performance improvement to `Request.content`.
-   Migrate to using collections.abc for 3.7 compatibility.

**Bugfixes**

-   Parsing empty `Link` headers with `parse_header_links()` no longer
    return one bogus entry.
-   Fixed issue where loading the default certificate bundle from a zip
    archive would raise an `IOError`.
-   Fixed issue with unexpected `ImportError` on windows system which do
    not support `winreg` module.
-   DNS resolution in proxy bypass no longer includes the username and
    password in the request. This also fixes the issue of DNS queries
    failing on macOS.
-   Properly normalize adapter prefixes for url comparison.
-   Passing `None` as a file pointer to the `files` param no longer
    raises an exception.
-   Calling `copy` on a `RequestsCookieJar` will now preserve the cookie
    policy correctly.

**Dependencies**

-   We now support idna v2.7.
-   We now support urllib3 v1.23.

2.18.4 (2017-08-15)
-------------------

**Improvements**

-   Error messages for invalid headers now include the header name for
    easier debugging

**Dependencies**

-   We now support idna v2.6.

2.18.3 (2017-08-02)
-------------------

**Improvements**

-   Running `$ python -m requests.help` now includes the installed
    version of idna.

**Bugfixes**

-   Fixed issue where Requests would raise `ConnectionError` instead of
    `SSLError` when encountering SSL problems when using urllib3 v1.22.

2.18.2 (2017-07-25)
-------------------

**Bugfixes**

-   `requests.help` no longer fails on Python 2.6 due to the absence of
    `ssl.OPENSSL_VERSION_NUMBER`.

**Dependencies**

-   We now support urllib3 v1.22.

2.18.1 (2017-06-14)
-------------------

**Bugfixes**

-   Fix an error in the packaging whereby the `*.whl` contained
    incorrect data that regressed the fix in v2.17.3.

2.18.0 (2017-06-14)
-------------------

**Improvements**

-   `Response` is now a context manager, so can be used directly in a
    `with` statement without first having to be wrapped by
    `contextlib.closing()`.

**Bugfixes**

-   Resolve installation failure if multiprocessing is not available
-   Resolve tests crash if multiprocessing is not able to determine the
    number of CPU cores
-   Resolve error swallowing in utils set\_environ generator

2.17.3 (2017-05-29)
-------------------

**Improvements**

-   Improved `packages` namespace identity support, for monkeypatching

requests/Scrape/last.py
import requests
import os
import json
from requests_oauthlib import OAuth1
from auth import *
from datetime import date, timedelta
import time
import sys

#SET URL AND AUTH
url = 'https://api.twitter.com/1.1/tweets/search/fullarchive/epitechdigital.json'
auth = OAuth1(CONSUMER_KEY, CONSUMER_SECRET, TOKEN, TOKEN_SECRET)

#GET IMPORTANT GLOBAL VAR
q = input("Query: ")
year = input("From: ")

#SET GLOBAL VARS
all_id = {}
nb_q = 0
nb_t = 0
theapi = {}
q_array_mrg = {}
#DELETE PREVIOUS QUERIES FROM Q



#SCRAPER
def alldays(year):
    dates = date(year, 1, 1)
    alldates = []
    while dates.year < year+1:
        alldates.append(dates)
        dates += timedelta(days = 7)
    return alldates

the_date = alldays(int(year))
for i in range(1):
    from_date = the_date[i]
    from_date = str(from_date).replace('-', '')
    to_date = the_date[i+1]
    to_date = str(to_date).replace('-', '')
    querys = requests.get(url, params={'query': q,' maxResults': 2, 'fromDate': from_date + '0000', 'toDate': to_date + '0000'}, auth=auth)
    f = open( 'birth' + '.json', 'a')
    f.write(querys.text)
    f.close
    """json_query = querys.json()['results']
    count_json = len(json_query)
    nb_q += count_json"""
    for remaining in range(2, 0, -1):
        sys.stdout.write("\r")
        sys.stdout.write("{:2d} seconds remaining.".format(remaining)) 
        sys.stdout.flush()
        time.sleep(1)

#ANSWER
"""print('La requête a permis de récolter ' + str(nb_q) + ' tweets')"""

requests/Scrape/main_two.py
import requests
import os
import json
from auth import *
# To set your enviornment variables in your terminal run the following line:
# export 'BEARER_TOKEN'='<your_bearer_token>'


def auth():
    return os.environ.get(bearer_token)


def create_url():
    query = "from:twitterdev -is:retweet"
    # Tweet fields are adjustable.
    # Options include:
    # attachments, author_id, context_annotations,
    # conversation_id, created_at, entities, geo, id,
    # in_reply_to_user_id, lang, non_public_metrics, organic_metrics,
    # possibly_sensitive, promoted_metrics, public_metrics, referenced_tweets,
    # source, text, and withheld
    tweet_fields = "tweet.fields=author_id"
    url = "https://api.twitter.com/2/tweets/search/recent?query={}&{}".format(
        query, tweet_fields
    )
    return url


def create_headers(bearer_token):
    headers = {"Authorization": "Bearer {}".format(bearer_token)}
    return headers


def connect_to_endpoint(url, headers):
    response = requests.request("GET", url, headers=headers)
    print(response.status_code)
    if response.status_code != 200:
        raise Exception(response.status_code, response.text)
    return response.json()


def main():
    bearer_token = auth()
    url = create_url()
    headers = create_headers(bearer_token)
    json_response = connect_to_endpoint(url, headers)
    print(json.dumps(json_response, indent=4, sort_keys=True))


if __name__ == "__main__":
    main()

requests/Scrape/Twitter_scrape.py
import requests
import os
import json
from requests_oauthlib import OAuth1
from auth import *

#SET URL AND AUTH
url = 'https://api.twitter.com/1.1/search/tweets.json'
auth = OAuth1(CONSUMER_KEY, CONSUMER_SECRET, TOKEN, TOKEN_SECRET)

#GET IMPORTANT GLOBAL VAR
q = input("Query: ")
nb_r = input("Request's number: ")

#GET QUERY
query = requests.get(url, params={'q': q, 'count': 50}, auth=auth)

#SET VAR MAX & MIN
max_id = query.json()['search_metadata']['max_id'] + 2500000000000000
min_id = query.json()['search_metadata']['since_id']
max = max_id
min = min_id

#SET GLOBAL VARS
all_id = {}
nb_t = 0
theapi = {}
q_array_mrg = {}
#DELETE PREVIOUS QUERIES FROM Q

#SCRAPER
for i in range(int(nb_r)):
    querys = requests.get(url, params={'q': q, 'count': 50, 'since_id': min, 'max_id': max}, auth=auth)
    if 'statuses' not in querys.json():
        max = max + 2500000000000000
        min = min + 2500000000000000
    else:
        json_query = querys.json()['statuses']
        count_json = len(json_query)
        if count_json == 0:
            max = max + 2500000000000000
            min = min + 2500000000000000
        else:
            for n in range(count_json):
                ids = querys.json()['statuses'][n]['id']
                all_id[i] = ids
            if count_json > 0:
                if i == 0:
                    print(query.json()['search_metadata']['max_id'])
                    nb_t += count_json                        
                    for n in range(count_json):
                        querys.json()['statuses'][n]['id']
                        id = querys.json()['statuses'][n]['id']
                        text = querys.json()['statuses'][n]['text']
                        created_at = querys.json()['statuses'][n]['created_at']
                        q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                    theapi['query'+str(i)] = q_array_mrg
                else:
                    print(query.json()['search_metadata']['max_id'])
                    im = i - 1
                    is_equal = all_id[im] == all_id[i]
                    if is_equal == True:
                        max = max + 2500000000000000
                        min = min + 2500000000000000
                    else:
                        nb_t += count_json
                        for n in range(count_json):
                            id = querys.json()['statuses'][n]['id']
                            text = querys.json()['statuses'][n]['text']
                            created_at = querys.json()['statuses'][n]['created_at']
                            q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                        theapi['query'+str(i)] = q_array_mrg
                        max = max + 2500000000000000
                        min = min + 2500000000000000

directory = q
path_dir = r"C:\Users\axelz\OneDrive\Documents\Epitech Digital\Projet\Storks\Storks\resources\json"
if not os.path.exists(path_dir + "\\"+ q):
    os.mkdir(os.path.join(path_dir, q))
f = open(os.path.join(path_dir + "\\"+ q, q + '.json'), 'w')
total = json.dumps(theapi)
f.write(total)
f.close

#ANSWER
print('La requête a permis de récolter ' + str(nb_t) + ' tweets')

requests/Scrape/main_request.py
import requests
import os
import json
from requests_oauthlib import OAuth1
from auth import *

"""def json_dumps(args):
    for arg in range(args):
        arg = """
from datetime import date, timedelta
day_name= ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday','Sunday']
def allsundays(year):
    d = date(year, 1, 1)                    # January 1st
    d += timedelta(days = 7 - d.weekday())  # First Sunday
    while d.year < year+1:
        yield d
        d += timedelta(days = 7)
 
def re(d):
    for i in range(2):
        querys = requests.get(url, params={'q': q, 'count': 50, 'until': d}, auth=auth)
        if 'statuses' not in querys.json():
            f = open( q + 'dbg.json', 'a')
            f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
            f.close
            break
        else:
            json_query = querys.json()['statuses']
            count_json = len(json_query)
            if count_json == 0:
                f = open( q + 'dbg.json', 'a')
                f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
                f.close
                break
            else:
                for n in range(count_json):
                    ids = querys.json()['statuses'][n]['id']
                    all_id[i] = ids
                if count_json > 0:
                    if i == 0:
                        print(query.json()['search_metadata']['max_id'])
                        nb_t += count_json                        
                        for n in range(count_json):
                            querys.json()['statuses'][n]['id']
                            id = querys.json()['statuses'][n]['id']
                            text = querys.json()['statuses'][n]['text']
                            created_at = querys.json()['statuses'][n]['created_at']
                            q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                        theapi['query'+str(i)] = q_array_mrg
                    else:
                        print(query.json()['search_metadata']['max_id'])
                        im = i - 1
                        is_equal = all_id[im] == all_id[i]
                        if is_equal == True:
                            f = open( q + 'dbg.json', 'a')
                            f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
                            f.close
                            break
                        else:
                            nb_t += count_json
                            for n in range(count_json):
                                id = querys.json()['statuses'][n]['id']
                                text = querys.json()['statuses'][n]['text']
                                created_at = querys.json()['statuses'][n]['created_at']
                                q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                            theapi['query'+str(i)] = q_array_mrg

#SET URL AND AUTH
url = 'https://api.twitter.com/1.1/tweets/search/fullarchive/epitech.json'
auth = OAuth1(CONSUMER_KEY, CONSUMER_SECRET, TOKEN, TOKEN_SECRET)

#GET IMPORTANT GLOBAL VAR
q = input("Query: ")
year = input("Year: ")

#GET QUERY
query = requests.get(url, params={'q': q, 'count': 50}, auth=auth)

#SET VAR MAX & MIN
"""max_id = query.json()['search_metadata']['max_id'] + 2500000000000000
min_id = query.json()['search_metadata']['since_id']"""
max = 0 #max_id
min = 0 #min_id

#SET GLOBAL VARS
all_id = {}
nb_t = 0
theapi = {}
q_array_mrg = {}
#DELETE PREVIOUS QUERIES FROM Q

#SCRAPER
for d in allsundays(int(year)):
    
    querys = requests.get(url, params={'query': q, 'maxResults': 100}, auth=auth)
    f = open( q + '.json', 'w')
    f.write(querys.text)
    f.close
    if 'statuses' not in querys.json():
        f = open( q + 'dbg.json', 'a')
        f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
        f.close       
        re(d)
    else:
        json_query = querys.json()['statuses']
        count_json = len(json_query)
        if count_json == 0:
            f = open( q + 'dbg.json', 'a')
            f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
            f.close           
            re(d)
        else:
            for n in range(count_json):
                ids = querys.json()['statuses'][n]['id']
                all_id[i] = ids
            if count_json > 0:
                if i == 0:
                    print(query.json()['search_metadata']['max_id'])
                    nb_t += count_json                        
                    for n in range(count_json):
                        querys.json()['statuses'][n]['id']
                        id = querys.json()['statuses'][n]['id']
                        text = querys.json()['statuses'][n]['text']
                        created_at = querys.json()['statuses'][n]['created_at']
                        q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                    theapi['query'+str(i)] = q_array_mrg
                else:
                    print(query.json()['search_metadata']['max_id'])
                    im = i - 1
                    is_equal = all_id[im] == all_id[i]
                    if is_equal == True:
                        f = open( q + 'dbg.json', 'a')
                        f.write(str(d) + ' re2.2 - ' + str(query.json()['search_metadata']['max_id']) + "\n")
                        f.close
                        re(d)
                    else:
                        nb_t += count_json
                        for n in range(count_json):
                            id = querys.json()['statuses'][n]['id']
                            text = querys.json()['statuses'][n]['text']
                            created_at = querys.json()['statuses'][n]['created_at']
                            q_array_mrg[n] = [{'id': id, 'text': text, 'created_at': created_at}]
                        theapi['query'+str(i)] = q_array_mrg
                        max = max + 2500000000000000
                        min = min + 2500000000000000

directory = q
path_dir = r"C:\Users\axelz\OneDrive\Documents\Epitech Digital\Projet\Storks\Storks\resources\json"
if not os.path.exists(path_dir + "\\"+ q):
    os.mkdir(os.path.join(path_dir, q))
f = open(os.path.join(path_dir + "\\"+ q, q + '.json'), 'w')
total = json.dumps(theapi)
f.write(total)
f.close

#ANSWER
print('La requête a permis de récolter ' + str(nb_t) + ' tweets') + "\n"